doc/nvgre_tunnel/nvgre_tunnel.md

```diff
@@ -0,0 +1,322 @@
+
+The new YANG model should be created in order to auto-generate CLI by using the [SONiC CLI Auto-generation tool](https://github.com/Azure/SONiC/blob/master/doc/cli_auto_generation/cli_auto_generation.md).
+
+### Requirements
+
+This section describes the SONiC requirements for NVGRE feature.
+
+#### Functional requirements
+
+At a high level the following should be supported:
+
+- Phase #1
+  - NVGRE tunnel should be able to work in parallel to VxLAN tunnel
+- Phase #2
+  - CLI for NVGRE tunnel
+
+#### Orchagent requirements
+
+NVGRE orchagent:
+
+- Phase #1
+  - Should be able to create Bridge/VLAN to VSID mapping
+  - Should be able to create tunnels and encap/decap mappers
+
+### Architecture Design 
+
+The new tables will be added to Config DB. Unless otherwise stated, the attributes are mandatory.
+
+#### NVGRE configDB table
+
+```
+NVGRE_TUNNEL|{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+
+NVGRE_TUNNEL_MAP|{{tunnel_name}}|{{tunnel_map}}
+    "vsid": {{vsid_id}}
+    "vlan": {{vlan_id}}
+```
+
+#### ConfigDB schemas
+
+```
+; Defines schema for NVGRE Tunnel configuration attributes
+key                                   = NVGRE_TUNNEL:name             ; NVGRE tunnel configuration
+; field                               = value
+SRC_IP                                = ipv4                          ; IPv4 source address
+
+;value annotations
+ipv4          = dec-octet "." dec-octet "." dec-octet "." dec-octet
+dec-octet     = DIGIT                     ; 0-9  
+                  / %x31-39 DIGIT         ; 10-99  
+                  / "1" 2DIGIT            ; 100-199  
+                  / "2" %x30-34 DIGIT     ; 200-249
+```
+
+```
+; Defines schema for NVGRE Tunnel map configuration attributes
+key                                   = NVGRE_TUNNEL:tunnel_name:name ; NVGRE tunnel configuration
+; field                               = value
+VSID                                  = DIGITS                        ; 1 to 16 million values
+VLAN                                  = 1\*4DIGIT                     ; 1 to 4094 Vlan id
+```
+
+#### Orchestration agent
+
+The following orchestration agents will be added or modified. The flow diagrams are captured in a later section.
+
+##### Figure 1. Orchestration agents
+
+<p align=center>
+<img src="images/nvgre_orch.svg" alt="Figure 1. Orchestration agents">
+</p>
+
+#### NvgreOrch
+
+`nvgreorch` - it is an orchestration agent that handles the configuration requests directly from ConfigDB. The `nvgreorch` is responsible for creates the tunnel and attaches encap and decap mappers. Separate tunnel maps are created for L3 NVGRE and can attach different VLAN/VSID or Bridge/VSID to a respective tunnel.
```

Comment:
is responsible for creates -> is responsible for creating ...

Reply:
Both VLAN and Bridge tunnel maps are created for NVGRE tunnel by default so it would be possible to apply both VLAN and Bridge map entries.